### ISSUE TITLE
IP address pings to 127.0.0.1 when getting disconnected

### ISSUE THREAD

**Issue by N0VANT0** [karashiiro/PingPlugin#43]
It seems like this issues has been happening since v2.0.2.0 and a version before that version. Every time I get disconnected, the plugin will ping 127.0.0.1 instead of the server, even switching and logging in on different datacenter doesn't fix it. The only way to fix it is to restart the game or disable/reenable the plugin in-game. I don't know which version it was, but the version during/that was used for 6.05  patch works without any problem (it evens pings the datacenter on the main menu without the need to log into your character). This issue still happens when I'm not using any VPN.

I'm not a developer so I am not able to provide any useful log files, but if it really needed, I will learn and try to generate one.

**Comment by N0VANT0** [comment]
Hi, you might be dreading to hear this but the the bug still persists with 2 variations:

1. I forcefully disconnect myself to get Error 90002 from the game by disabling my network adapter, log back in after enabling my network adapter on the same Data Center and the plugin will not ping the server and instead still keep pinging 127.0.0.1. Switching Data Center (logging in with my Materia character) will make it work again. This is the log that I managed to get when I try to log back in after getting Error 90002 (note that I am not including the line where the plugin manages to work correctly, only the error logs).

--------------------------------
07:43:42.436 | ERR Unobserved exception in Task.
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (An exception occurred during a Ping request.)
 ---> System.Net.NetworkInformation.PingException: An exception occurred during a Ping request.
 ---> System.ComponentModel.Win32Exception (1231): The network location cannot be reached. For information about network troubleshooting, see Windows Help.
   at System.Net.NetworkInformation.Ping.GetStatusFromCode(Int32 statusCode)
   at System.Net.NetworkInformation.Ping.DoSendPingCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options, Boolean isAsync)
   at System.Net.NetworkInformation.Ping.SendPingAsyncInternal(IPAddress address, Int32 timeout, Byte[] buffer, PingOptions options)
   --- End of inner exception stack trace ---
   at System.Net.NetworkInformation.Ping.SendPingAsyncInternal(IPAddress address, Int32 timeout, Byte[] buffer, PingOptions options)
   at PingPlugin.PingTrackers.ComponentModelPingTracker.PingLoop(CancellationToken token) in K:\arashiiro\PingTrackers\ComponentModelPingTracker.cs:line 23
   --- End of inner exception stack trace ---


2. I log out momentarily to turn on my vpn (mudfish with fastconnect mode) after playing the game without vpn, the plugin did not detect that I've logged into my elemental Data Center and keeps pinging 127.0.0.1. This is the log that I managed to get.

07:58:07.190 | ERR Unobserved exception in Task.
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at PingPlugin.PingTrackers.PingTracker.UpdateSeAddress() in K:\arashiiro\PingTrackers\PingTracker.cs:line 150
   at PingPlugin.PingTrackers.PingTracker.AddressUpdateLoop(CancellationToken token)
   --- End of inner exception stack trace ---
--------------------------------
07:58:07.199 | ERR Unobserved exception in Task.
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at PingPlugin.PingTrackers.PingTracker.UpdateSeAddress() in K:\arashiiro\PingTrackers\PingTracker.cs:line 150
   at PingPlugin.PingTrackers.PingTracker.AddressUpdateLoop(CancellationToken token)
   --- End of inner exception stack trace ---
--------------------------------
07:58:07.199 | ERR Unobserved exception in Task.
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at PingPlugin.PingTrackers.PingTracker.UpdateSeAddress() in K:\arashiiro\PingTrackers\PingTracker.cs:line 150
   at PingPlugin.PingTrackers.PingTracker.AddressUpdateLoop(CancellationToken token)
   --- End of inner exception stack trace ---
{ } [ Send ]

I can do some more testing if you want (even with video with dalamud log open).
